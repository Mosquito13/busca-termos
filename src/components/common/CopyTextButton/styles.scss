@import '../../../sass/styles.scss';

.copy-text-button {
    width: 2rem;
    display: flex;
    align-items: center;
    margin: 0 1rem;
    color: currentColor;
    transition: transform 0.2s;
    cursor: pointer;

    &--no-margin-right {
        margin-right: 0;
    }

    &:hover {
        @include themify() {
            color: themed('color-copy-text-button-hover');
        }
    }

    &:active {
        transform: translateY(1px);
    }
}