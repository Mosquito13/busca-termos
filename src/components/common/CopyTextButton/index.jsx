import { useCallback } from 'react';
import { FiCopy } from 'react-icons/fi';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Icon, { SMALL } from '../Icon';

import './styles.scss';

const CopyTextButton = ({ noMarginRight, value }) => {
  const classes = classNames(
    'copy-text-button',
    noMarginRight && 'copy-text-button--no-margin-right'
  );

  const handleCopy = useCallback(() => {
    window.navigator.clipboard.writeText(value);
  }, [value]);

  return (
    <div className={classes} onClick={handleCopy} title="Copiar">
      <Icon size={SMALL}>
        <FiCopy />
      </Icon>
    </div>
  );
};

CopyTextButton.propTypes = {
  /**
   * Valor para ser copiado
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Se esconde a margem direita
   */
  noMarginRight: PropTypes.bool
};

export default CopyTextButton;
