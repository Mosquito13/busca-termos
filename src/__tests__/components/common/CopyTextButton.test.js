import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CopyTextButton from '../../../components/common/CopyTextButton';

describe(`${CopyTextButton.name}`, () => {
  describe('tests', () => {
    let wrapper;

    beforeAll(() => {
      Object.assign(global.navigator, {
        clipboard: {
          writeText: jest.fn()
        }
      });

      wrapper = shallow(<CopyTextButton value="A value" />);
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.copy-text-button')).toHaveLength(1);
    });

    it('should set value to clipboard when click', () => {
      wrapper.find('.copy-text-button').simulate('click');
      expect(global.navigator.clipboard.writeText).toBeCalled();
      expect(global.navigator.clipboard.writeText).toBeCalledWith('A value');
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render with margin right', () => {
      const tree = renderer.create(<CopyTextButton value="abc" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render without margin right', () => {
      const tree = renderer
        .create(<CopyTextButton noMarginRight value="abc" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
