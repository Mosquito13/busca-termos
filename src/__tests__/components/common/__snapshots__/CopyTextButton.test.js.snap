// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyTextButton snapshots should render with margin right 1`] = `
<div
  className="copy-text-button"
  onClick={[Function]}
  title="Copiar"
>
  <div
    className="icon icon--small"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="13"
        rx="2"
        ry="2"
        width="13"
        x="9"
        y="9"
      />
      <path
        d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
      />
    </svg>
  </div>
</div>
`;

exports[`CopyTextButton snapshots should render without margin right 1`] = `
<div
  className="copy-text-button copy-text-button--no-margin-right"
  onClick={[Function]}
  title="Copiar"
>
  <div
    className="icon icon--small"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="13"
        rx="2"
        ry="2"
        width="13"
        x="9"
        y="9"
      />
      <path
        d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
      />
    </svg>
  </div>
</div>
`;
